src_features/performPrivacyOperation/cmd_performPrivacyOperation.c

```diff
@@ -0,0 +1,122 @@
+#include "shared_context.h"
+#include "apdu_constants.h"
+
+#include "ui_flow.h"
+#include "feature_performPrivacyOperation.h"
+
+#define P2_PUBLIC_ENCRYPTION_KEY 0x00
+#define P2_SHARED_SECRET         0x01
+
+void decodeScalar(const uint8_t *scalarIn, uint8_t *scalarOut) {
+    for (uint8_t i = 0; i < 32; i++) {
+        switch (i) {
+            case 0:
+                scalarOut[0] = (scalarIn[31] & 0x7f) | 0x40;
+                break;
+            case 31:
+                scalarOut[31] = scalarIn[0] & 0xf8;
+                break;
+            default:
+                scalarOut[i] = scalarIn[31 - i];
+        }
+    }
```

Comment:
But I don't understand why to make a switch case that matches 0 and 31 because there is a for from 0 to 32 so necessarily at each turn we check 2 conditions that we are sure to catch

why not do something like this
```c
for (uint8_t i = 0; i < 32; i++) {
  scalarOut[i] = scalarIn[31 - i];
}
scalarOut[0] = (scalarIn[31] & 0x7f) | 0x40;
scalarOut[31] = scalarIn[0] & 0xf8;
```

Reply:
no specific reason - it does exactly the same thing